python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -323,36 +392,161 @@ def filterReferences(self, exposure, refCat, refWcs):
         expPolygon = lsst.sphgeom.ConvexPolygon(expSkyCorners)
 
         # Step 2: Filter out reference catalog sources that are
-        # not contained within the exposure boundaries.
-        sources = type(refCat)(refCat.table)
-        for record in refCat:
-            if expPolygon.contains(record.getCoord().getVector()):
-                sources.append(record)
-        refCatIdDict = {ref.getId(): ref.getParent() for ref in sources}
-
-        # Step 3: Cull sources that do not have their parent
-        # source in the filtered catalog.  Save two copies of each
-        # source.
-        refSources = type(refCat)(refCat.table)
-        for record in refCat:
-            if expPolygon.contains(record.getCoord().getVector()):
-                recordId = record.getId()
-                topId = recordId
-                while (topId > 0):
-                    if topId in refCatIdDict:
-                        topId = refCatIdDict[topId]
-                    else:
-                        break
-                if topId == 0:
-                    refSources.append(record)
-
-        # Step 4: Transform source footprints from the reference
-        # coordinates to the exposure coordinates.
-        for refRecord in refSources:
-            refRecord.setFootprint(refRecord.getFootprint().transform(refWcs,
-                                                                      expWcs, expRegion))
-        # Step 5: Replace reference catalog with filtered source list.
-        return refSources
+        # not contained within the exposure boundaries, or whose
+        # parents are not within the exposure boundaries.  Note
+        # that within a single input refCat, the parents always
+        # appear before the children.
+        mergedRefCat = lsst.afw.table.SourceCatalog(refCats[0].table)
+        for refCat in refCats:
+            containedIds = {0}  # zero as a parent ID means "this is a parent"
+            for record in refCat:
+                if expPolygon.contains(record.getCoord().getVector()) and record.getParent() in containedIds:
+                    record.setFootprint(record.getFootprint().transform(refWcs, expWcs, expRegion))
```

Comment:
This is resurrected from an old commit, but from what I researched, if there is no footprint (or its zero area) then there is no coord associated with it (as none can be measured. This will cause `getCoord` and `getVector` to create special `NaN` associated objects. One of these objects is never contained inside any sphere geom regions, therefore this record is skipped.